raiden/network/proxies/token_network.py

```diff
@@ -1071,81 +1067,66 @@ def _update_preconditions(
-            msg = (
-                'updateNonClosingBalanceProof cannot be called '
-                'because the settlement period is over'
-            )
-            raise RaidenRecoverableError(msg)
         else:
-            error_type, msg = self._check_channel_state_for_update(
-                channel_identifier=channel_identifier,
-                closer=partner,
-                update_nonce=nonce,
-                block_identifier=block_identifier,
-            )
-            if error_type:
-                raise error_type(msg)
+            if our_recovered_address != self.node_address:
+                raise RaidenUnrecoverableError('Invalid balance proof signature')
 
-    def update_transfer(
-            self,
-            channel_identifier: ChannelID,
-            partner: Address,
-            balance_hash: BalanceHash,
-            nonce: Nonce,
-            additional_hash: AdditionalHash,
-            closing_signature: Signature,
-            non_closing_signature: Signature,
-            given_block_identifier: BlockSpecification,
-    ):
-        log_details = {
-            'token_network': pex(self.address),
-            'node': pex(self.node_address),
-            'partner': pex(partner),
-            'nonce': nonce,
-            'balance_hash': encode_hex(balance_hash),
-            'additional_hash': encode_hex(additional_hash),
-            'closing_signature': encode_hex(closing_signature),
-            'non_closing_signature': encode_hex(non_closing_signature),
-        }
-        log.debug('updateNonClosingBalanceProof called', **log_details)
+            if partner_recovered_address != partner:
+                raise RaidenUnrecoverableError('Invalid update transfer signature')
 
-        checking_block = self.client.get_checking_block()
+        # Check the preconditions for calling updateNonClosingBalanceProof at
+        # the time the event was emitted.
         try:
-            self._update_preconditions(
+            channel_onchain_detail = self._detail_channel(
+                participant1=self.node_address,
+                participant2=partner,
+                block_identifier=given_block_identifier,
                 channel_identifier=channel_identifier,
-                partner=partner,
-                balance_hash=balance_hash,
-                nonce=nonce,
-                additional_hash=additional_hash,
-                closing_signature=closing_signature,
+            )
+            closer_details = self._detail_participant(
+                channel_identifier=channel_identifier,
+                participant=partner,
+                partner=self.node_address,
                 block_identifier=given_block_identifier,
             )
-        except NoStateForBlockIdentifier:
-            # If preconditions end up being on pruned state skip them. Estimate
-            # gas will stop us from sending a transaction that will fail
+        except ValueError:
```

Comment:
This try-except is handling pruned data. If first call may succeed and the second can fail things will still work (IOW, the preconditions checks are skipped).